packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2029,41 +2047,67 @@ class RenderPhysicalShape extends _RenderPhysicalModelBase<Path> {
 
   @override
   void paint(PaintingContext context, Offset offset) {
```

Comment:
This version looks identical to the one above. Can we make this a common helper method in the base class?

Reply:
Hmm, lots of fiddly bits different between the implementations. Turning them all into paths might omit optimizations under Skia (though Impeller might be less impacted by that).

Reply:
Yeah these have been split up for a while. If we were to refactor them I would think we'd split up the clipping into its own RO or remove these altogether.